cli/src/org/partiql/shell/Shell.kt

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package org.partiql.shell
+
+import com.amazon.ion.system.IonTextWriterBuilder
+import com.amazon.ionelement.api.toIonValue
+import com.google.common.base.CharMatcher
+import com.google.common.util.concurrent.Uninterruptibles
+import org.fusesource.jansi.AnsiConsole
+import org.jline.reader.EndOfFileException
+import org.jline.reader.LineReader
+import org.jline.reader.LineReaderBuilder
+import org.jline.reader.UserInterruptException
+import org.jline.reader.impl.completer.NullCompleter
+import org.jline.terminal.TerminalBuilder
+import org.jline.utils.AttributedStringBuilder
+import org.jline.utils.AttributedStyle
+import org.jline.utils.InfoCmp
+import org.joda.time.Duration
+import org.partiql.lang.CompilerPipeline
+import org.partiql.lang.eval.Bindings
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.lang.eval.delegate
+import org.partiql.lang.syntax.Parser
+import org.partiql.lang.util.ConfigurableExprValueFormatter
+import org.partiql.lang.util.ExprValueFormatter
+import java.io.Closeable
+import java.io.OutputStream
+import java.io.PrintStream
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.Locale
+import java.util.Properties
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.annotation.concurrent.GuardedBy
+
+private const val PROMPT_1 = "PartiQL> "
+private const val PROMPT_2 = "   | "
```

Comment:
`PROMPT_2` is never used.

Reply:
Let me update so PROMPT_2 is used as well as multiple newlines. I do not get that warning. Have you tried running the executable script rather than via Gradle? Does that change anything?

Reply:
Yeah, just checked via the executable, and it works perfectly. History works smoothly, and the prompt `>` works.

Running via Gradle gets rid of all the ability to use the history and the prompt.

Reply:
Yeah, just checked via the executable, and it works perfectly. History works smoothly, and the prompt `>` works.

Running via Gradle gets rid of all the ability to use the history and the prompt.

Reply:
Yep. I left a comment about this in `docs/user/CLI.md`